index.js

```diff
@@ -1,341 +1,390 @@
-			},
-			{
-				type: 'textinput',
-				id: 'IP',
-				label: 'Desk IP',
-				regex: this.REGEX_IP,
-				width: 12
-			}
-		]
-	}
+    this.FIELD_PERCENTAGE = {
+      type: 'number',
+      label: 'Percentage (0->100)',
+      id: 'percentage',
+      default: 100,
+      min: 0,
+      max: 100
+    }
 
-	/**
-	 * Clean up the instance before it is destroyed.
-	 *
-	 * @access public
-	 * @since 1.0.0
-	 */
-	destroy() {
-		if (this.pollInterval !== undefined) {
-			clearInterval(this.pollInterval);
-		}
+    this.FIELD_PLAYBACKACTION = {
+      type: 'dropdown',
+      label: 'ON/OFF',
+      id: 'playbackaction',
+      default: '0',
+      choices: [
+        { label: 'ON', id: '0' },
+        { label: 'OFF', id: '1' }
+      ]
+    }
 
-		this.debug("destroy", this.id);
-	}
+    this.FIELD_USERNUMBER = {
+      type: 'number',
+      label: 'UserNumber',
+      id: 'un',
+      default: 1,
+      min: 1,
+      max: 9999
+    }
 
-	/**
-	 * INTERNAL: Get an image from the console for a user number
-	 *
-	 * @param {number} un - the user number
-	 * @param {string} icon - URL of the icon to get
-	 * @since 1.1.0
-	 */
-	getImage(un, icon) {
-		this.request.get(icon, (error, response, body) => {
-			if (!error && response.statusCode == 200) {
-				this.imageList[un] = Buffer.from(body).toString('base64');
-			}
-		});
-	}
+    this.FIELD_ALWAYSREFIRE = {
+      type: 'checkbox',
+      label: 'Always Refire',
+      id: 'refire',
+      default: true
+    }
 
-	/**
-	 * Main initialization function called once the module
-	 * is OK to start doing things.
-	 *
-	 * @access public
-	 * @since 1.0.0
-	 */
-	init() {
+    this.updateActions(); // export actions
+		this.initFeedbacks();
 		this.sendCommand("get/System/SoftwareVersion");
-	}
+  }
 
-	/**
-	 * INTERNAL: initialize feedbacks.
-	 *
-	 * @access protected
-	 * @since 1.1.0
-	 */
-	initFeedbacks() {
-		var feedbacks = {};
+  /**
+   * INTERNAL: initialize feedbacks.
+   *
+   * @access protected
+   * @since 1.1.0
+   */
+  initFeedbacks () {
+    var feedbacks = {}
 
-		feedbacks['image'] = {
-			label: 'Get picture legend of playback',
-			description: 'Updates the keys image to the image on your console',
-			options: [
-				this.FIELD_USERNUMBER
-			],
-			callback: (feedback, bank) => {
-				if (this.imageList[feedback.options.un] !== undefined) {
-					return {
-						png64: this.imageList[feedback.options.un]
-					}
-				}
-				else {
-					return {
-						png64: "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII="
-					}
-				}
-			}
-		};
+    feedbacks['image'] = {
+      type: 'advanced',
+      name: 'Get picture legend of playback',
+      description: 'Updates the keys image to the image on your console',
+      options: [this.FIELD_USERNUMBER],
+      callback: (feedback, bank) => {
+        if (this.imageList[feedback.options.un] !== undefined) {
+          return {
+            png64: this.imageList[feedback.options.un]

```

Comment:
`png64` is no longer a return type.  This should be:
``imageBuffers: [{buffer: this.imageList[feedback.options.un }]``